### PR TITLE
Increase coverage for S3 list and read

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Detail
* Added tests to increase coverage
* Removed some unused code

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 8f7d488beab27eefd2a1fafbbb1fa89d45158920
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d2b7d66d-8ec8-4d74-969f-898da0c30010%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 8f7d488beab27eefd2a1fafbbb1fa89d45158920
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0d26b702-e226-43f5-b7b5-1ab55622b75f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 8f7d488beab27eefd2a1fafbbb1fa89d45158920
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fe0e8ebd-dc5a-4f60-9bb0-d4c0bc6b2b7e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]

File: awswrangler/s3/_read.py [hunk 1]
```diff
@@ -90,48 +87,15 @@ def _apply_partitions(df: pd.DataFrame, dataset: bool, path: str, path_root: Opt
...
+def _union(dfs: List[pd.DataFrame], ignore_index: bool) -> pd.DataFrame:
     cats: Tuple[Set[str], ...] = tuple(set(df.select_dtypes(include="category").columns) for df in dfs)
     for col in set.intersection(*cats):
         cat = union_categoricals([df[col] for df in dfs])
         for df in dfs:
             df[col] = pd.Categorical(df[col].values, categories=cat.categories)
     return pd.concat(objs=dfs, sort=False, copy=False, ignore_index=ignore_index)
-
-
-def _read_dfs_from_multiple_paths(
```
Comment (kukushking) [inline awswrangler/s3/_read.py:115]: Nice, thanks for cleaning this up! Indeed looks like with the executor we no longer use it

**Review by kukushking — APPROVED** [review]

**Pull request merged**